dali/kernels/imgproc/color_manipulation/equalize/hist_test.cc

```diff
@@ -0,0 +1,137 @@
+      auto num_channels = in_view.shape[sample_idx][1];
+      out_batch_shape.set_tensor_shape(sample_idx, {num_channels, range_size});
+    }
+    baseline_.reshape(out_batch_shape);
+    out_.reshape(out_batch_shape);
+    auto out_view = out_.cpu();
+    // fill output with non-zero value to make sure that the kernel does not rely on zeroed memory
+    for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+      for (int64_t i = 0; i < out_view[sample_idx].num_elements(); i++) {
+        out_view[sample_idx].data[i] = 3;
+      }
+    }
+    auto baseline_view = baseline_.cpu();
+    for (int sample_idx = 0; sample_idx < baseline_view.shape.num_samples(); sample_idx++) {
+      auto sample_baseline_view = baseline_view[sample_idx];
+      auto sample_in_view = in_view[sample_idx];
+      auto num_channels = sample_in_view.shape[1];
+      ASSERT_EQ(num_channels, sample_baseline_view.shape[0]);
+      for (int64_t idx = 0; idx < sample_baseline_view.num_elements(); idx++) {
+        sample_baseline_view.data[idx] = 0;
+      }
+      for (int64_t pixel_idx = 0; pixel_idx < sample_in_view.shape[0]; pixel_idx++) {
+        for (int64_t channel_idx = 0; channel_idx < num_channels; channel_idx++) {
+          auto val = sample_in_view.data[pixel_idx * num_channels + channel_idx];
+          ++sample_baseline_view.data[channel_idx * range_size + val];
+        }
+      }
+    }
+  }
+
+  void PrepareUniformRandom(TensorListShape<2> batch_shape) {
+    in_.reshape(batch_shape);
+    UniformRandomFill(in_.cpu(), rng_, 0, 255);
+  }
+
+  TestTensorList<uint8_t, 2> in_;
+  TestTensorList<uint64_t, 2> baseline_, out_;
+  std::mt19937_64 rng_{12345};
+};
+
+TEST_F(EqualizeHistGpuTest, UniformRandChannel1Batch1) {
+  this->PrepareUniformRandom(TensorListShape<2>{{{1024 * 1024 - 7, 1}}});
+  this->Run();
+}
+
+TEST_F(EqualizeHistGpuTest, UniformRandChannel3Batch1) {
+  this->PrepareUniformRandom(TensorListShape<2>{{{511 * 307, 3}}});
+  this->Run();
+}
+
+TEST_F(EqualizeHistGpuTest, UniformRandVarChannels) {
+  TensorListShape<2> tls{
+      {{1, 1}, {101, 7}, {205, 4}, {511 * 702, 2}, {4096, 4}, {4096, 4}, {4096 - 5, 11}}};
+  this->PrepareUniformRandom(tls);
+  this->Run();
+}
+
+TEST_F(EqualizeHistGpuTest, SingleValue) {
+  TensorListShape<2> batch_shape{{{1024 * 1024 * 64, 1},
+                                  {1024 * 1024 * 64 - 1, 2},
+                                  {1024 * 1024 * 64 - 2, 3},
+                                  {1024 * 1024 * 64 - 3, 4},
+                                  {1024 * 1024 * 64 - 4, 5},
+                                  {1024 * 1024 * 64 - 4, 6}}};
+  in_.reshape(batch_shape);
+  auto in_view = in_.cpu();
+  for (int sample_idx = 0; sample_idx < in_view.num_samples(); sample_idx++) {
+    auto sample_in_view = in_view[sample_idx];
+    for (int64_t idx = 0; idx < batch_shape[0].num_elements(); idx++) {
+      sample_in_view.data[idx] = 51 * sample_idx;
+    }
+  }
```

Comment:
You are not running the test, just setting the data. Is that expected?

Reply:
Right, I missed it, thanks.